### PR TITLE
WINDUPRULE-226 Rule for Detecting Hibernate and JSF embedded library

### PR DESCRIPTION
**Pull request by mrizzi**

### CODE REVIEW

**Comment by windupgithubbot1** [comment]
Can one of the admins verify this patch?

**Comment by mareknovotny** [comment]
great job with your first java rule ;) @mrizzi !

Just one minor thing I have, could you split the rules into dedicated classes - one for JSF and one for Hibernate?

**Comment by mrizzi** [comment]
thanks @mareknovotny: i had great tutors ;)
I'm going to split it into 2 rules.

File: rules-java-ee/addon/src/main/java/org/jboss/windup/rules/apps/javaee/rules/DiscoverEmbeddedHibernateLibraryRuleProvider.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+    @Override
+    public Configuration getConfiguration(RuleLoaderContext context)
+    {
+        String ruleIDPrefix = getClass().getSimpleName();
+        int ruleIDSuffix = 1;
+        return ConfigurationBuilder.begin()
+                    .addRule()
+                    .when(Query.fromType(JarArchiveModel.class)
+                                .withProperty(FileModel.FILE_PATH, QueryPropertyComparisonType.REGEX, ".*WEB-INF/lib/.*hibernate.*\\.jar$"))
```
Comment (mrizzi) [inline rules-java-ee/addon/src/main/java/org/jboss/windup/rules/apps/javaee/rules/DiscoverEmbeddedHibernateLibraryRuleProvider.java]: I think i'm going to make the rule searching in `EAR/lib` path too.

File: rules-java-ee/addon/src/main/java/org/jboss/windup/rules/apps/javaee/rules/DiscoverEmbeddedJSFLibraryRuleProvider.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+    @Override
+    public Configuration getConfiguration(RuleLoaderContext context)
+    {
+        String ruleIDPrefix = getClass().getSimpleName();
+        int ruleIDSuffix = 1;
+        return ConfigurationBuilder.begin()
+                    .addRule()
+                    .when(Query.fromType(JarArchiveModel.class)
+                                .withProperty(FileModel.FILE_PATH, QueryPropertyComparisonType.REGEX, ".*WEB-INF/lib/.*jsf.*\\.jar$"))
```
Comment (mrizzi) [inline rules-java-ee/addon/src/main/java/org/jboss/windup/rules/apps/javaee/rules/DiscoverEmbeddedJSFLibraryRuleProvider.java]: Just as above ;)

File: rules-java-ee/addon/src/main/java/org/jboss/windup/rules/apps/javaee/rules/DiscoverEmbeddedHibernateLibraryRuleProvider.java (hunk 1 above)
Comment (jsight) [inline rules-java-ee/addon/src/main/java/org/jboss/windup/rules/apps/javaee/rules/DiscoverEmbeddedHibernateLibraryRuleProvider.java]: I'm not sure that it should be restricted by location at all. The path for JAR files tends to be configurable and ear-libs aren't always only in EAR/lib.

Comment (jsight) [inline rules-java-ee/addon/src/main/java/org/jboss/windup/rules/apps/javaee/rules/DiscoverEmbeddedHibernateLibraryRuleProvider.java]: Yes

Comment (jsight) [inline rules-java-ee/addon/src/main/java/org/jboss/windup/rules/apps/javaee/rules/DiscoverEmbeddedHibernateLibraryRuleProvider.java]: I realize that introduces a small possibility that they might have them somewhere in the archive that is not actually picked up by the application server. That seems like a small risk, though, compared to potentially missing competitive server's preferred location (I think some use APP-LIB or something like that).

**Pull request merged**